File: 1561-141-1400-construct-k-palindrome-strings/1561-141-1400-construct-k-palindrome-strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        if (k > s.length()) {
            return false;
        }

        int freq = 0;
        for (char ch : s.toCharArray()) {
            freq ^= (1 << (ch - 'a'));
        }

        return Integer.bitCount(freq) <= k;
    }
}

File: 1561-141-1400-construct-k-palindrome-strings/1561-141-1400-construct-k-palindrome-strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        if(k > s.size())return false;
        int frq = 0; 
        for(char ch : s)
            frq ^= (1<<(ch-'a'));      
        return __builtin_popcount(frq) <= k;
    }
};